static/third/bootstrap-typeahead/typeahead.js

```diff
@@ -428,6 +428,12 @@
   , click: function (e) {
       e.stopPropagation()
       e.preventDefault()
+      var target = e.currentTarget && e.currentTarget.tagName.toLowerCase() === "li" ? e.currentTarget: e.target
+      if (target && target.tagName.toLowerCase() !== "li"){
+        // If it's still not our desired element it might be its child.
+          target = $(target).closest("li")
+          this.mouseenter({currentTarget: target})
+      }
```

Comment:
Just fix the delegation above and you won’t have to write so much spaghetti logic.

```diff
       this.$menu
-        .on('click', this.click.bind(this))
+        .on('click', 'li', this.click.bind(this))
         .on('mouseenter', 'li', this.mouseenter.bind(this))
```

Reply:
Yeah, This is much better. Thanks Anders.  
Updated the PR!